src/engine_registry.cpp

```diff
@@ -444,441 +442,441 @@ namespace wr
 	}();
 
 	REGISTER(state_objects::state_object, RTPipelineRegistry)(
-		{
-			StateObjectDescription::D3D12StateObjectDesc(D3D12_STATE_OBJECT_TYPE_RAYTRACING_PIPELINE),
-			StateObjectDescription::Library(rt_full_lib),
-			StateObjectDescription::MaxPayloadSize((sizeof(float) * 7) + sizeof(unsigned int)),
-			StateObjectDescription::MaxAttributeSize(sizeof(float) * 4),
-			StateObjectDescription::MaxRecursionDepth(3),
-			StateObjectDescription::GlobalRootSignature(root_signatures::rt_test_global),
-			StateObjectDescription::LocalRootSignatures(std::nullopt),
-		});
+	{
+		.desc = D3D12_STATE_OBJECT_TYPE_RAYTRACING_PIPELINE,
+		.library_desc = rt_full_lib,
+		.max_payload_size = (sizeof(float) * 7) + sizeof(unsigned int),
+		.max_attributes_size = sizeof(float) * 4,
+		.max_recursion_depth = 3,
+		.global_root_signature = root_signatures::rt_test_global,
+		.local_root_signatures = {},
+	});
 
 	/* ### Depth of field ### */
 
 	// Cone of confusion
 	REGISTER(shaders::dof_coc, ShaderRegistry) ({
-		ShaderDescription::Path("resources/shaders/dof_coc.hlsl"),
-			ShaderDescription::Entry("main_cs"),
-			ShaderDescription::Type(ShaderType::DIRECT_COMPUTE_SHADER)
-		});
+		.path = "resources/shaders/dof_coc.hlsl",
+		.entry = "main_cs",
+		.type = ShaderType::DIRECT_COMPUTE_SHADER
```

Comment:
Perhaps make this variable more explicit? Like ".shader_type", just a suggestion